scanner/src/main/kotlin/scanners/fossid/FossIdConfig.kt

```diff
@@ -132,6 +141,7 @@ internal data class FossIdConfig(
             val addAuthenticationToUrl = fossIdScannerOptions[CREDENTIALS_IN_URL_PROPERTY].toBoolean()
             val waitForResult = fossIdScannerOptions[WAIT_FOR_RESULT_PROPERTY]?.toBooleanStrictOrNull() ?: true
             val deltaScans = fossIdScannerOptions[DELTA_SCAN_PROPERTY].toBoolean()
+            val deltaScanLimit = fossIdScannerOptions[DELTA_SCAN_LIMIT_PROPERTY]?.toInt() ?: Int.MAX_VALUE
```

Comment:
You should enforce `deltaScanLimit` > 0 (it's noticeable where you use it and do `-1`).

Reply:
So, what semantics does `Int.MAX_VALUE` have here? Does it mean "unlimited", or does it really mean the numeric value of `Int.MAX_VALUE`?

What I'm after is: I'm not so happy with "misusing" numeric values that are within the valid range for something special, like the meaning of "unlimited"; I would have rather said a value of `0` means "unlimited", if `0` does not make sense to be use otherwise.

Reply:
It means "unlimited". This may be a difference from a logic point of view, but in practice this can be treated as equivalent (as we will never reach this number of delta scans). Using this high value has the advantage that no conditional logic is needed when enforcing this limit.

But if you insist in changing this, I would rather use a *null* value as undefined, hoping that some of Kotlin's constructs would make the necessary distinctions more straight-forward.

Reply:
While I have to admit that I don't like this mix of semantics, I do see the value of handling it this way for code simplicity. So let's keep it.